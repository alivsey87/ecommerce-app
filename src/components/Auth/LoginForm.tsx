import { useState } from "react";
import { signInWithEmailAndPassword } from "firebase/auth";
import { auth } from "../../firebase/firebaseConfig";
import './login.css';

type LoginFormProps = {
  onClose: () => void;
};

const LoginForm = ({ onClose }: LoginFormProps) => {
  const [email, setEmail] = useState<string>("");
  const [password, setPassword] = useState<string>("");
  const [error, setError] = useState<string | null>(null);

  const handleLogin = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    setError(null);
    try {
      await signInWithEmailAndPassword(auth, email, password);

      alert("Login successful!");
      onClose();
    } catch (err: unknown) {
      setError(err instanceof Error ? err.message : "Unknown error occurred.");
    }
  };

  return (
    <>
      <form className="main-login-form" onSubmit={handleLogin}>
        <h3 className="form-head">Login</h3>
        <input
          type="email"
          placeholder="Email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
        />
        <input
          type="password"
          placeholder="Password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
        />
        <button className="btn-main" type="submit">Login</button>
        {error && <p>{error}</p>}
      </form>
    </>
  );
};

export default LoginForm;
