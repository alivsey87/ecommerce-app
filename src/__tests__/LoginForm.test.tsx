import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import LoginForm from "../components/Auth/LoginForm";


const renderLoginForm = () => {
  return render(<LoginForm onClose={() => {}} />);
};

describe("LoginForm", () => {
  it("renders form elements", () => {
    renderLoginForm();

    expect(screen.getByRole("heading", { name: "Login" })).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Email")).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
    expect(screen.getByRole("button", { name: "Login" })).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = renderLoginForm();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("updates input values when typing", () => {
    renderLoginForm();

    const emailInput = screen.getByPlaceholderText("Email");
    const passwordInput = screen.getByPlaceholderText("Password");

    fireEvent.change(emailInput, { target: { value: "test@example.com" } });
    fireEvent.change(passwordInput, { target: { value: "password123" } });

    expect(emailInput).toHaveValue("test@example.com");
    expect(passwordInput).toHaveValue("password123");
  });
});
