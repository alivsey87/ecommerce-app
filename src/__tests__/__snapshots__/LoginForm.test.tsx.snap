// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches snapshot 1`] = `
<form
  class="main-login-form"
>
  <h3
    class="form-head"
  >
    Login
  </h3>
  <input
    placeholder="Email"
    type="email"
    value=""
  />
  <input
    placeholder="Password"
    type="password"
    value=""
  />
  <button
    class="btn-main"
    type="submit"
  >
    Login
  </button>
</form>
`;
